modules/core/shared/src/main/scala/io/circe/Codec.scala

```diff
@@ -0,0 +1,38 @@
+package io.circe
+
+/**
+  *
+  * Convenience typeclass to be used when you need both an [[io.circe.Encoder]] and
+  * [[io.circe.Decoder]] for your type.
+  *
+  * @author Lorand Szakacs, lsz@lorandszakacs.com
+  * @since 21 Dec 2017
+  *
+  */
+trait Codec[A] extends Encoder[A] with Decoder[A]
+
+object Codec {
+  def apply[A](implicit instance: Codec[A]): Codec[A] = instance
+
+  def instance[A](encode: Encoder[A], decode: Decoder[A]): Codec[A] =
+    new Codec[A] {
+      private val enc = encode
```

Comment:
What's the benefit of having these private vals? `apply` implementations could use the parameters directly